ci/travis/ci.sh

```diff
@@ -127,24 +313,90 @@ build() {
     eval "$(curl -sL https://raw.githubusercontent.com/travis-ci/gimme/master/gimme | GIMME_GO_VERSION=stable bash)"
   fi
 
-  if [ "${LINUX_WHEELS-}" = 1 ]; then
-    # Mount bazel cache dir to the docker container.
-    # For the linux wheel build, we use a shared cache between all
-    # wheels, but not between different travis runs, because that
-    # caused timeouts in the past. See the "cache: false" line below.
-    local MOUNT_BAZEL_CACHE=(-v "${HOME}/ray-bazel-cache":/root/ray-bazel-cache -e TRAVIS=true -e TRAVIS_PULL_REQUEST="${TRAVIS_PULL_REQUEST}" -e encrypted_1c30b31fe1ee_key="${encrypted_1c30b31fe1ee_key-}" -e encrypted_1c30b31fe1ee_iv="${encrypted_1c30b31fe1ee_iv-}")
+  case "${OSTYPE}" in
```

Comment:
No need to check OSTYPE here, we just need to make sure that `LINUX_WHEELS`, `MAC_WHEELS` and `WINDOWS_WHEELS` are only defined on the right OS.

Reply:
In terms of the design I actually think it should be the other way around; the redundant bit is those flags. (They should just be just a single `WHEELS` flag.)

I can look into doing it the other way around, though I'm not sure how feasible it is, since it depends somewhat on what kinds of conditoinal expressions we can put as the definitions of these variables.

Reply:
Yeah so like I expected it doesn't seem like there's a reasonable way to do this in the YAML file since e.g. `LINUX_WHEELS: ${{ runner.os == 'Linux' }` turns into `LINUX_WHEELS=true` rather than `LINUX_WHEELS=1`. Not sure how you want me to do it otherwise, it seems any other way would be less elegant?

Reply:
Why not just set ``WHEELS_TYPE: ${{ runner.os }}`` and then check for that here?

Reply:
Well I was also trying not to modify how the existing code works at the same time as I'm setting up GHA, but that solution also seems redundant and different from how we handle other flags? (And it suggests the wheel type can be different from the current OS which would be misleading.)